lib/pure/uri.nim

```diff
@@ -311,6 +356,46 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+proc parseQuery*(query: string): OrderedTableRef[string, string] = 
+  ## parses the query part of an uri.
+  var start = 0
+  var key, val = ""
+  var queryBlock = ""
+  if query == "":
+    return
+  result = newOrderedTable[string, string]()
+  for i in 0..query.len:
+    if query[i] == '&' or i == query.len:
+      start = i
+      var inval = false
+      for each in queryBlock:
+        if not inval and each == '=': 
+          inval = true
+          continue
+        if inval:
+          val.add each
+        else:
+          key.add each
+      result.add(key.decodeUrl, val.decodeUrl)
+      setLen(key, 0)
+      setLen(val, 0)
+      setLen(queryBlock, 0)
+    else:
+      queryBlock.add(query[i])
+
+proc parseQuery*(uri: Uri): OrderedTableRef[string, string] =
+  return parseQuery(uri.query)
+  
+proc newQuery(query: OrderedTableRef[string, string]): string =
```

Comment:
Create a new proc called ``newQuery`` which will take ``{"foo": "bar", ...}`` and create a new ``Query`` out of it.

(Example of its use shown [here](https://github.com/nim-lang/Nim/pull/5477/files#r104933439))